raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -246,20 +264,30 @@ private List<ReplicaState> followersByDescendingFetchOffset() {
             .collect(Collectors.toList());
     }
 
-    private boolean updateEndOffset(ReplicaState state,
-                                    LogOffsetMetadata endOffsetMetadata) {
+    private void verifyEndOffsetUpdate(
+        ReplicaState state,
+        LogOffsetMetadata endOffsetMetadata
+    ) {
         state.endOffset.ifPresent(currentEndOffset -> {
             if (currentEndOffset.offset > endOffsetMetadata.offset) {
                 if (state.nodeId == localId) {
                     throw new IllegalStateException("Detected non-monotonic update of local " +
-                        "end offset: " + currentEndOffset.offset + " -> " + endOffsetMetadata.offset);
+                            "end offset: " + currentEndOffset.offset + " -> " + endOffsetMetadata.offset);
                 } else {
                     log.warn("Detected non-monotonic update of fetch offset from nodeId {}: {} -> {}",
-                        state.nodeId, currentEndOffset.offset, endOffsetMetadata.offset);
+                            state.nodeId, currentEndOffset.offset, endOffsetMetadata.offset);
                 }
             }
         });
-
+    }
+    private boolean updateEndOffset(
+        ReplicaState state,
+        LogOffsetMetadata endOffsetMetadata,
+        boolean verifyUpdate
```

Comment:
Why do we need this? Seems like `false` is the only value we ever pass.

Reply:
Wanted to retain the verification step in the method in case we want to use it elsewhere in the future.